internal/controller/pdbwatcher_controller.go

```diff
@@ -105,11 +105,20 @@ func (r *PDBWatcherReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, err
 	}
 
+	// Track initial state to detect conflicts
+	initialResourceVersion := deployment.ResourceVersion
+	initialReplicas := *deployment.Spec.Replicas
+
 	// Update status dynamically
-	pdbWatcher.Status.CurrentReplicas = *deployment.Spec.Replicas
-	pdbWatcher.Status.MinReplicas = *deployment.Spec.Replicas
-	pdbWatcher.Status.MaxReplicas = *deployment.Spec.Replicas + deployment.Spec.Strategy.RollingUpdate.MaxSurge.IntVal
-	pdbWatcher.Status.ScaleFactor = 1 // Dynamically Calculated
+	if pdbWatcher.Status.CurrentReplicas == 0 {
+		pdbWatcher.Status.CurrentReplicas = initialReplicas
+	}
+	pdbWatcher.Status.MinReplicas = initialReplicas
+	if deployment.Spec.Strategy.RollingUpdate != nil && deployment.Spec.Strategy.RollingUpdate.MaxSurge != nil {
+		pdbWatcher.Status.MaxReplicas = initialReplicas + int32(deployment.Spec.Strategy.RollingUpdate.MaxSurge.IntValue())
```

Comment:
need to deal with string value which could be a precent. (this si pretty common)